src/node-file-trace.ts

```diff
@@ -152,8 +158,10 @@ export class Job {
   async readlink (path: string) {
     const cached = this.symlinkCache.get(path);
     if (cached !== undefined) return cached;
+    await this.fileIOQueue.acquire();
     try {
       const link = await fsReadlink(path);
+      await this.fileIOQueue.release();
```

Comment:
These [aren't](https://github.com/vercel/async-sema/blob/603842b82e68e24b76eb0e74dae66de1eda468b4/src/index.ts#L195-L197) async operations.

```suggestion
      this.fileIOQueue.release();
```